flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/declarative/allocator/SharedSlotTest.java

```diff
@@ -166,6 +168,48 @@ public void testReleaseForbidsSubsequentLogicalSlotAllocations() {
         sharedSlot.allocateLogicalSlot();
     }
 
+    @Test
+    public void testCanReturnLogicalSlotDuringRelease() {
+        final TestingPhysicalSlot physicalSlot = TestingPhysicalSlot.builder().build();
+        final SharedSlot sharedSlot =
+                new SharedSlot(new SlotRequestId(), physicalSlot, false, () -> {});
+        final LogicalSlot logicalSlot1 = sharedSlot.allocateLogicalSlot();
+        final LogicalSlot logicalSlot2 = sharedSlot.allocateLogicalSlot();
+
+        // both slots try to release the other one, simulating that the failure of one execution due
+        // to the release also fails others
+        logicalSlot1.tryAssignPayload(
+                new TestLogicalSlotPayload(
+                        cause -> {
+                            if (logicalSlot2.isAlive()) {
+                                logicalSlot2.releaseSlot(cause);
+                            }
+                        }));
+        logicalSlot2.tryAssignPayload(
+                new TestLogicalSlotPayload(
+                        cause -> {
+                            if (logicalSlot1.isAlive()) {
+                                logicalSlot1.releaseSlot(cause);
+                            }
+                        }));
+
+        sharedSlot.release(new Exception("test"));
```

Comment:
What are we asserting here?

Reply:
That the release did not fail with an exception; I'll add some assertios to make this more explicit